packages/report-server/src/reportBuilder/transform/parser/TransformParser.ts

```diff
@@ -22,8 +22,12 @@ type RowLookup = {
  */
 type Lookups = {
   row: Row;
+  previous: Row;
+  next: Row;
   all: RowLookup;
   allPrevious: RowLookup;
+  index: number; // one-based index, this.currentRow + 1
```

Comment:
Curious how you guys feel about maybe moving the `index` to be a field on a row, rather than an individual lookup here. That way it would be referenced like so: `$row.index` instead of `$index`

To me it seems to have:
Pros:
- Allows for some funky lookup behaviour, eg. 
`"'Increase on past 3 weeks'": 'row.value - sum(where(otherRow =  row.index - 3 <= otherRow.index and otherRow.index < row.index)).value'` 
- Makes a bit of sense to me, as we see the 'row #' in the table in the UI preview in the viz builder, so the user is already thinking of it as a 'column' in a way (subject I know)

Cons:
- Could introduce column naming clashes... I've tried really hard to avoid these so far, since we even support dynamic column names based on data from survey results, so ensuring that a clash doesn't occur seems _**very**_ hard...

Happy with current implementation as it stands though, it meets our needs and I don't know if I've come down off the fence on this issue yet... However if either of you are swayed to go with the `$row.index` implementation perhaps it's best if we take that route now?

Reply:
I'm pretty hesitant to add anything to the row that might plausibly be used as a column name.

I see the value of having the index on the row for those past three weeks type references...but I also think that a) you could already solve it by using a transform that applies $index to each row, giving you all the powers you have above, b) we could make an alias to do point a if it was a common case, and c) we could support the same behaviour in other ways that might be even more friendly than using a `where` (like a future $useRow($index - 3) or something...)

This has made me think though...is `$index` itself quite a dev-centric term? Should we instead be using a more excel-user friendly term like `$rowNumber`?

Reply:
I think for consistency I prefer `$row.something` to always be a literal column reference. And like Edwin says, it's very easy to add index to the columns.

```
{
  transform: 'select',
  "'index'": '$index',
},
```

Also if you wanna get into the weeds `INDEX` is a function in excel to perform a lookup in a table, so they _should_ be familiar with the concept, but also this is definitely me not wanting to keep making changes 😅

Reply:
Haha fair enough, we've lumped enough changes on you already...happy to keep `$index`!

Reply:
Great points @edmofro and @EMcQ-BES I'm happy with keeping it as `$index` too! 

> Should we instead be using a more excel-user friendly term like $rowNumber?

I actually had this thought too when we were first coming up with this, but when I did a bit of a google it seemed like `index` was actually the most commonly used term as @EMcQ-BES mentions!